test/explicit-length-check.js

```diff
@@ -0,0 +1,65 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/explicit-length-check';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errors = [{
+	ruleId: 'explicit-length-check',
+	message: '`length` property should be compared to a value'
+}];
+
+ruleTester.run('explicit-length-check', rule, {
+	valid: [
+		'array.foo',
+		'array.length',
+		'array.length === 0',
+		'array.length !== 0',
+		'array.length > 0',
+		'if (array.foo) {}',
+		'if (length) {}',
+		'if ([].length > 0) {}',
+		'if ("".length > 0) {}',
+		'if (array.length === 0) {}',
+		'if (array.length !== 0) {}',
+		'if (array.length !== 0 && array[0] === 1) {}'
+	],
+	invalid: [
+		{
+			code: 'if ([].length) {}',
+			errors
```

Comment:
> Sounds good. I'm still a bit torn on this, do you feel like creating an issue?

Feel free, but I'm personally done creating issues on ESLint. My views are just too incompatible with theirs and I don't have the energy or time for endless discussions that usually ends up with them closing it as won't fix.